udemy_exercises/cell_phone/CellPhone.java

```diff
@@ -1,19 +1,94 @@
-package udemy_exercises.cell_phone;
+package cell_phone;
+
+import java.util.ArrayList;
 
 class CellPhone {
+	
+	private ArrayList<Contact> contacts;
+
+	public CellPhone() {
+		this.contacts = new ArrayList<Contact>();
+	}
+	
+	public boolean addContact(Contact contact) {
+		if (findContact(contact.getName()) >= 0) {
+			System.out.println("Contact is already on file.");
+			return false;
+		}
+		contacts.add(contact);
+		return true;
+	}
+
+	private int findContact(Contact contact) {
+		return this.contacts.indexOf(contact);
+	}
+
+	private int findContact(String contactName) {
+		for (int i = 0; i < this.contacts.size(); i++) {
+			Contact contact = this.contacts.get(i);
+			if (contact.getName().equals(contactName)) {
+				return i;
+			}
+		}
+		return -1;
+	}
+
+	public boolean updateContact(Contact oldContact, Contact newContact) {
+		int foundPosition = this.findContact(oldContact);
+		if (foundPosition > 0) {
```

Comment:
Is there more than one "bad" value that can be provided by this.findContact(oldContact)?  If not, consider checking for that specific value here, and returning 'false' in this code block, moving this code to the next code block.